src/logion/services/authentication.service.ts

```diff
@@ -0,0 +1,97 @@
+import jwt, { Algorithm } from "jsonwebtoken";
+import { injectable } from "inversify";
+import { Express, Request, Response, NextFunction } from "express";
+import { StrategyOptions, ExtractJwt, Strategy } from "passport-jwt";
+import passport from "passport";
+import { Log } from "../util/Log";
+
+const AUTH_SECRET = "secret-key-that-should-never-be-disclosed-to-anybody-especially-the-bad-guys";
+const AUTH_ISSUER = "dev.logion.network";
+const TTL = 3600;
+const ALGORITHM: Algorithm = "HS384";
+
+const { logger } = Log;
+
+interface LogionUser {
+    account: string,
+    legalOfficer: boolean;
+}
+
+@injectable()
+export class AuthenticationService {
+
+    private configureEndpoints(app: Express, authenticate: any, authorizeForLegalOfficer: any): void {
+
+        app.all("/api/protection-request/*", authenticate)
+        app.post("/api/protection-request/:id/accept", authenticate, authorizeForLegalOfficer)
+        app.post("/api/protection-request/:id/reject", authenticate, authorizeForLegalOfficer)
+
+        app.all("/api/token-request/*", authenticate)
+        app.post("/api/token-request/:id/accept", authenticate, authorizeForLegalOfficer)
+        app.post("/api/token-request/:id/reject", authenticate, authorizeForLegalOfficer)
+        app.post("/api/token-request/:id/asset", authenticate, authorizeForLegalOfficer)
+
+        app.all("/api/transaction/*", authenticate)
+    }
+
+    createToken(address: string): string {
+        const now = Math.floor(Date.now() / 1000);
+        const payload = {
+            iat: now,
+            exp: now + TTL,
+            legalOfficer: true
+        };
+        return jwt.sign(payload, this.secret(), {
+            algorithm: ALGORITHM,
+            issuer: AUTH_ISSUER,
+            subject: address
+        });
+    }
+
+    private secret(): Buffer {
+        return Buffer.from(AUTH_SECRET, 'base64');
+    }
+
+    configureAuthentication(app: Express): void {
```

Comment:
Move to another service (configuration of authentication is not authentication)?

Reply:
Done